rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -2254,6 +2250,22 @@ public JobIdData submit(@HeaderParam("sessionid") String sessionId,
         }
     }
 
+    private String normalizeFilePath(String filePath, String fileName) {
+        if (filePath == null) {
+            filePath = "";
+        }
+        if (fileName != null && filePath.length() > 0 && !filePath.endsWith("/")) {
+            filePath = (filePath + "/" + fileName);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "filePath". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1226)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove those useless parentheses. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessParenthesesCheck)